lib/manageiq/ui/classic/engine.rb

```diff
@@ -29,7 +29,7 @@ class Engine < ::Rails::Engine
         #
         #   https://bugs.ruby-lang.org/issues/14372
         #
-        config.autoload_paths << root.join('app', 'controllers', 'mixins').to_s
+        config.autoload_paths << root.join('app', 'controllers').to_s
```

Comment:
Thought rails auto added `app/*` but I guess since this is an engine, we may need to add the paths.

Reply:
thanks, I'll double check

Reply:
I don't understand this change.  The mixins directory was intentionally added to act like the "concerns" folders that are typically autoloaded and treated like any other autoload directory.

Reply:
@Fryguy I think the problem was we had inconsistency with some things requiring or autoloading using `Mixins::` namespace and others not.  I was trying to consolidate them to all use the same way.  What you see here is one of those ways: changes things to use `Mixins::` namespace.

When I tried going the route of "use mixins" autoload_paths (no `Mixins::` on include), I had weird things like this happening:

Autoloading using [CustomButtons](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/mixins/custom_buttons.rb#LL1C1-L1C1)(strip Mixins from autoload) seemed very confusing to me as that conflicts mentally with [CustomButton](https://github.com/ManageIQ/manageiq/blob/master/app/models/custom_button.rb).

I felt the more intentional thing was to use the `Mixins` namespace as the other way would require more work to move/name things to be more intentional and unique.

There are other examples such as [MiqPolicyMixin](https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/miq_policy_mixin.rb) in core and [PolicyMixin](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/mixins/policy_mixin.rb).

Reply:
Ideally I would expect mixins to _not_ be namespaced similar to how we do other concerns (e.g. we don't do Concerns::Xxx), so I would expect mixins dir in the load path.

That being said if it's too much work right now then we have to choose what's the most expedient way forward and we can always rename the mixins in the future, so I'm fine with this. Let's open a follow up issue to "fix" these mixins to not need the namespace and to put mixins on the load path (perhaps in a concerns directory instead of mixins for consistency), since, IMO, that's the least surprising layout.